tensordict/tensordict.py

```diff
@@ -1830,10 +1844,14 @@ def flatten_keys(
                     inner_tensordict = self.get(key).flatten_keys(
                         separator=separator, inplace=inplace
                     )
-                    for inner_key, inner_item in inner_tensordict.items():
-                        tensordict_out.set(separator.join([key, inner_key]), inner_item)
+                    if inner_tensordict is not self.get(key):
```

Comment:
I wonder if this is compatible with lazy tensordicts (eg `LazyStackedTensorDict`) where two calls to `get(key)` return items which ids are different (eg because they are the results of a call to `stack`)

Reply:
It's interesting point. I didn't research 'LazyStackedTensorDict' 's behaviour. But if 'get(key)' return new ids every time how can we detect self loop?